### PR TITLE
Add contract badge

### PR DESCRIPTION
**Pull request by nachoiacovino**

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:



- [**/api/badges/contract**](https://thirdweb-6wucbmjxw-thirdweb.vercel.app/api/badges/contract)



Commit 8e98e67c6039fec670f6ffabdef7c419514da9ad (https://thirdweb-6wucbmjxw-thirdweb.vercel.app).

File: @3rdweb-sdk/react/useContractBadge.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    const res = await fetch(`/api/badges/contract`, {
+      method: "POST",
+      body: JSON.stringify({ address }),
+    });
+    if (res.status >= 400) {
+      throw new Error(await res.json().then((r) => r.error));
+    }
+    return res.json() as Promise<any>;
+  });
+}
```
Comment (jnsdls) [inline @3rdweb-sdk/react/useContractBadge.ts:14]: ideally we don't need this and instead we can just make a normal GET request to the api route with query param of chainId + contract address -> it returns the svg as an image (so then u can put it in an  for example)

File: components/badges/contract-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const contractBadge = useContractBadge(contractAddress);
+
+  console.log(contractBadge);
+
+  if (!contractBadge?.data?.svg) {
+    return null;
+  }
+
+  return <div>{parse(contractBadge.data.svg)}</div>;
+};
```
Comment (jnsdls) [inline components/badges/contract-badge.tsx:25]: nah, don't need this

File: package.json [hunk 3]
```diff
@@ -43,6 +43,7 @@
     "ethers": "^5.4.6",
     "flat": "^5.0.2",
     "framer-motion": "^7.0.0",
+    "html-react-parser": "^3.0.4",
```
Comment (jnsdls) [inline package.json:46]: pls no

File: pages/api/badges/contract.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  if (req.method !== "POST") {
+    return res.status(400).json({ error: "Please use POST method" });
+  }
+
+  const { address } = JSON.parse(req.body);
+
+  try {
+    const fontBuffer = await readFile(
+      resolve("./public/assets/fonts", "Inter-medium.ttf"),
+    );
```
Comment (jnsdls) [inline pages/api/badges/contract.tsx:20]: for addresses we use monospace so might want monospace font here? (IBM plex mono...)

File: pages/api/badges/contract.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            height: "18px",
+            width: "28px",
+          }}
+        >
+          <img
+            src="https://thirdweb.com/brand/thirdweb-icon.png"
+            alt=""
+            width={28}
+            height={18}
+          />
```
Comment (jnsdls) [inline pages/api/badges/contract.tsx:48]: prob want a nice square size & `objectFit="contain"`

File: pages/api/badges/contract.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            name: "Inter",
+            data: fontBuffer,
+            weight: 400,
+            style: "normal",
+          },
+        ],
+      },
+    );
+
+    return res.status(200).json({ svg });
```
Comment (jnsdls) [inline pages/api/badges/contract.tsx:75]: nah this needs to return an svg image type not json, this endpoint should generate an svg always, never a JSON response

**Review by jnsdls — COMMENTED** [review]
good start, can clean a lot of it up tho

File: pages/api/badges/contract.tsx (hunk 4 above)
Comment (nachoiacovino) [reply pages/api/badges/contract.tsx:20]: Following the designs here that uses Inter for both the address and the "Audited" text, I like how it looks

**Review by jnsdls — CHANGES_REQUESTED** [review]
it now builds but it doesn't work at runtime... need to dig in further

**Comment by nachoiacovino** [comment]
Superseded by #611

**Pull request closed without merging**